inc/Functions/BuildModuleFilePath.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Filesystem\Path;
+
+ /**
+  * Builds the correct path to include a file from a particular module.
+  *
+  * $module and $file should only contain letters, numbers and valid chars like _ or -.
+  * If this is not the case, an exception will be thrown.
+  *
+  * This function also protects for path traversal attacks.
+  */
+function BuildModuleFilePath(Filesystem $filesystem, string $rootDirectory, string $module, string $file): string
+{
+    // Some files contain a `_` or `-` char.
+    // To avoid a regular expression, we allow the chars by replacing.
+    $validChars = array('-', '_');
+
+    // Check that they only contain valid chars like letters and numbers
```

Comment:
We may also want to prevent access to certain files (or potentially even file types).
Prevent Writes
* .htaccess
* user.ini

Prevent Reads and Writes
* config.php
* .git-credentials

Would it be a good idea to add handling for that?